include/GafferScene/Shader.h

```diff
@@ -158,6 +159,11 @@ class GAFFERSCENE_API Shader : public Gaffer::ComputeNode
 		/// cases.
 		virtual IECore::DataPtr parameterValue( const Gaffer::Plug *parameterPlug ) const;
 
+		const Gaffer::ValuePlug *parameterSource(
```

Comment:
Should be private. The other protected stuff is all for derived classes to override, but we're adding something with the intention that only the ShaderPlug can get to it (via friendship).